### PR TITLE
Don't reject URIs with encoded '/' with servlet 6

### PR DESCRIPTION
**Pull request by idlewis**
first part of the fix for #23570

### CODE REVIEW

**Comment by idlewis** [comment]
#build

**Comment by idlewis** [comment]
#build

**Comment by idlewis** [comment]
#build

**Comment by idlewis** [comment]
#build

**Comment by idlewis** [comment]
Previous build timed out.
new build:
https://wasrtc.hursley.ibm.com:9443/jazz/web/projects/WS-CD#action=com.ibm.team.build.viewResult&id=_tgFE4IHfEe257ttQcLfLow

**Comment by idlewis** [comment]
#libby-bot

**Comment by idlewis** [comment]
#build

**Comment by posmith** [comment]
L2 message review completed

**Comment by idlewis** [comment]
#build

File: dev/com.ibm.ws.webcontainer/resources/OSGI-INF/l10n/metatype.properties [hunk 1]
```diff
@@ -185,4 +185,7 @@ setHtmlContentTypeOnError.name=Disable text/html content type on error responses
 setHtmlContentTypeOnError.desc=When this property is false, the webcontainer will not set the response's content type header during the error handling process. An application is responsible to set the response's content type. The default value (true) sets the content type to "text/html".
 
 excludeAllHandledTypesClasses.name=Disable returning handlesTypes parameters
-excludeAllHandledTypesClasses.desc=When this property is set to true, during startup, ServletContainerInitializer implementors that use the HandlesTypes annotation do not receive classes that are specified as HandlesTypes parameters.
\ No newline at end of file
+excludeAllHandledTypesClasses.desc=When this property is set to true, during startup, ServletContainerInitializer implementors that use the HandlesTypes annotation do not receive classes that are specified as HandlesTypes parameters.
+
+skipEncodedCharVerification.name=Skip verification of encoded characters in URI
```
Comment (dmuelle) [inline dev/com.ibm.ws.webcontainer/resources/OSGI-INF/l10n/metatype.properties:190]:
```suggestion
skipEncodedCharVerification.name=Skip verification of encoded characters in the request URI.
```

File: dev/com.ibm.ws.webcontainer/resources/com/ibm/ws/webcontainer/resources/Messages.nlsprops [hunk 2]
```diff
@@ -1,5 +1,5 @@
 ###############################################################################
-# Copyright (c) 1997, 2020 IBM Corporation and others.
+# Copyright (c) 1997, 2022 IBM Corporation and others.
```
Comment (dmuelle) [inline dev/com.ibm.ws.webcontainer/resources/com/ibm/ws/webcontainer/resources/Messages.nlsprops:2]: should this be 2023?

File: dev/com.ibm.ws.webcontainer/resources/com/ibm/ws/webcontainer/resources/Messages.nlsprops [hunk 3]
```diff
@@ -1614,3 +1614,8 @@ duplicate.samesite.attribute.for.cookie.name.useraction=Check the deployment des
 duplicate.reported.for.samesite.attribute.cookie.name=CWWWC0004W: A {0} cookie name that is configured for the {1} SameSite attribute value was previously identified as a duplicate. No SameSite attribute is added for this cookie.
 duplicate.reported.for.samesite.attribute.cookie.name.explanation=A cookie name cannot be configured for multiple SameSite attribute values. This cookie name has already been identified as a duplicate.
 duplicate.reported.for.samesite.attribute.cookie.name.useraction=Check the deployment descriptor configuration to make sure the cookie name is configured for only one of the following SameSite attribute values: 'SameSiteCookies_Lax', 'SameSiteCookies_None', or 'SameSiteCookies_Strict'. 
+
+bad.request.uri\:.{0}=CWWWC0005I: The request URI has invalid or improperly encoded characters: [{0}]
```
Comment (dmuelle) [inline dev/com.ibm.ws.webcontainer/resources/com/ibm/ws/webcontainer/resources/Messages.nlsprops:1618]: is [{0}] a list of encoded characters? If so, 

```suggestion
bad.request.uri\:.{0}=CWWWC0005I: The request URI includes the following invalid or improperly encoded characters: [{0}]
```

also, avoid using brackets on the variable unless needed for readability

**Review by dmuelle — CHANGES_REQUESTED** [review]
ID review, a couple minor suggestions

File: dev/com.ibm.ws.webcontainer/resources/com/ibm/ws/webcontainer/resources/Messages.nlsprops (hunk 3 above)
Comment (dmuelle) [reply dev/com.ibm.ws.webcontainer/resources/com/ibm/ws/webcontainer/resources/Messages.nlsprops:1618]: Sorry- just noticed this is a draft PR, I'm removing my request for changes so as not to block it prematurely

**Review by dmuelle — APPROVED** [review]

**Pull request closed without merging**